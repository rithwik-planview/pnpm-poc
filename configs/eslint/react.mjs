import web from './web.mjs';
import react from 'eslint-plugin-react';
import reactHooks from 'eslint-plugin-react-hooks';

export default [
    ...web,
    react.configs.flat['jsx-runtime'],
    reactHooks.configs['recommended-latest'],
];
