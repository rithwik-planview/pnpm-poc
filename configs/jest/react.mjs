import base from './base.mjs';

export default {
    ...base,
    testEnvironment: 'jest-environment-jsdom',
};
