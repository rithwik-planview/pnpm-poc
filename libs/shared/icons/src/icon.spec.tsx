import { Copy } from './index';
import { render } from '@testing-library/react';
import 'jest-styled-components';

const snapshot = (str: string, ui: Parameters<typeof render>[0]) => {
    test(`${str}`, () => {
        const { container } = render(ui);
        expect(container.firstChild).toMatchSnapshot();
    });
};

describe('Unity icons', () => {
    snapshot('Icon with default container', <Copy />);
    snapshot('Icon with no container', <Copy size={16} />);
    snapshot('Icon with color', <Copy color="red" />);
});
