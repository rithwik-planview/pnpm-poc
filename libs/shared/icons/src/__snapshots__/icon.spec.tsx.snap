// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PV Icons Icon with color 1`] = `
.c0 {
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  height: 36px;
  min-height: 36px;
  width: 36px;
}

<div
  class="c0"
>
  <svg
    aria-hidden="true"
    color="red"
    data-testid="svg-icon-copy"
    fill="none"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10.284 2.652c.732 0 1.326.594 1.326 1.327v10.61c0 .732-.593 1.326-1.326 1.326H2.326A1.326 1.326 0 0 1 1 14.589V3.979c0-.733.594-1.327 1.326-1.327zM2.326 14.59h7.958V3.979H2.326zM13.6 0c.366 0 .663.297.663.663V12.6a.663.663 0 0 1-1.326 0V1.326H4.315a.663.663 0 0 1 0-1.326z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`PV Icons Icon with default container 1`] = `
.c0 {
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  height: 36px;
  min-height: 36px;
  width: 36px;
}

<div
  class="c0"
>
  <svg
    aria-hidden="true"
    data-testid="svg-icon-copy"
    fill="none"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10.284 2.652c.732 0 1.326.594 1.326 1.327v10.61c0 .732-.593 1.326-1.326 1.326H2.326A1.326 1.326 0 0 1 1 14.589V3.979c0-.733.594-1.327 1.326-1.327zM2.326 14.59h7.958V3.979H2.326zM13.6 0c.366 0 .663.297.663.663V12.6a.663.663 0 0 1-1.326 0V1.326H4.315a.663.663 0 0 1 0-1.326z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`PV Icons Icon with no container 1`] = `
<svg
  aria-hidden="true"
  data-testid="svg-icon-copy"
  fill="none"
  height="16"
  viewBox="0 0 16 16"
  width="16"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M10.284 2.652c.732 0 1.326.594 1.326 1.327v10.61c0 .732-.593 1.326-1.326 1.326H2.326A1.326 1.326 0 0 1 1 14.589V3.979c0-.733.594-1.327 1.326-1.327zM2.326 14.59h7.958V3.979H2.326zM13.6 0c.366 0 .663.297.663.663V12.6a.663.663 0 0 1-1.326 0V1.326H4.315a.663.663 0 0 1 0-1.326z"
    fill="currentColor"
  />
</svg>
`;
